### PR TITLE
grass7: bump version to 7.8.4 with added +app variant

### PR DESCRIPTION
**Pull request by nilason**
#### Description

This PR bumps GRASS GIS (grass7) to version 7.8.4.

In addition, it adds a `+app` variant to create/install a GRASS.app bundle to $applications_dir.

Other notable changes:
- the file `patch-Platform_make_in.diff` has been dropped, as it wasn't used anyway
- setting environment variable `GRASS_PYTHON` – now hardcoded to selected Python with a patch

###### Type(s)

- [ ] bugfix
- [x] enhancement
- [ ] security fix

###### Tested on

macOS 10.15.7 19H2
Xcode 12.0.1 12A7300

###### Verification 
Have you

- [x] followed our [Commit Message Guidelines](https://trac.macports.org/wiki/CommitMessages)?
- [x] squashed and [minimized your commits](https://guide.macports.org/#project.github)?
- [x] checked that there aren't other open [pull requests](https://github.com/macports/macports-ports/pulls) for the same change?
- [ ] referenced existing tickets on [Trac](https://trac.macports.org/wiki/Tickets) with full URL? 
- [x] checked your Portfile with `port lint`?
- [ ] tried existing tests with `sudo port test`?
- [ ] tried a full install with `sudo port -vst install`?
- [x] tested basic functionality of all binary files?

Update:
Modified PR to include app bundling into +gui variant.

### CODE REVIEW

File: gis/grass7/Portfile [hunk 1]
```diff
@@ -176,6 +177,50 @@ variant mysql5 description {Add MySQL 5 support} {
...
+        reinplace "s|@GRASS_VERSION_MINOR@|${grass_version_minor}|" ${info_plist}
+        reinplace "s|@GRASS_VERSION_DATE@|${grass_version_date}|" ${info_plist}
+        reinplace "s|@BUNDLE_VERSION@|${bundle_version}|" ${info_plist}
+        reinplace "s|@DEPLOYMENT_TARGET@|${macosx_deployment_target}|" ${info_plist}
+    }
+
+    post-destroot {
+        copy ${worksrcpath}/${app_name} ${destroot}${applications_dir}
+    }
+}
```
Comment (mf2k) [inline gis/grass7/Portfile:222]: Why is this a variant? Just always run this code.

**Review by mf2k — CHANGES_REQUESTED** [review]

File: gis/grass7/Portfile (hunk 1 above)
Comment (nilason) [reply gis/grass7/Portfile:222]: GRASS GIS may be used only as a library (e.g. for gdal-grass and qgis3), and there may be power users not interested in the gui at all. Therefore it is useful to have +app as a variant. I can imagine it as a default though!

Comment (mf2k) [reply gis/grass7/Portfile:222]: Then they don't have to use the GUI. We prefer to build in everything where feasible. Especially as this does not add any dependencies.

Comment (nilason) [reply gis/grass7/Portfile:222]: Well, it does add dependency.+app requires +gui, which in turn depends on the wxWidgets port.

Having that said, if you think the added dependency is reasonable I have no problem going all-in for gui and app.

Comment (mf2k) [reply gis/grass7/Portfile:222]: So what is the point of a `+gui` variant if it does not add an app? So how about adding this code to the existing `+gui `variant?

Comment (nilason) [reply gis/grass7/Portfile:222]: Up to this point, the gui had to be started through terminal.

However, good point: adding the app bundle stuff to the gui variant makes sense.

**Review by mf2k — APPROVED** [review]

**Comment by nilason** [comment]
Thanks!

**Pull request merged**